### ISSUE TITLE
stop storing Goal statements into modules

### ISSUE THREAD

**Issue by Alizter** [coq/coq#17442]
>I am fine with the way the examples are written. (I suppose that, in the long run, we should stop storing `Goal` statements into modules, which would solve this mess in a more intuitive way.)

_Originally posted by @silene in https://github.com/coq/coq/issues/17205#issuecomment-1479274964_

**Comment by Alizter** [comment]
I understood this as:
```Goal foo.
asdf.
Qed.
```
produces a definition with a generated name. We should discard these from the global environment completely when closing a module. (Or perhaps even when Qed).

**Comment by samuelgruetter** [comment]
(nitpick first: I think the title of this issue should say "stop *storing* Goal statements into modules", not "stop solving")

I often use `Goal` right after a `Definition` or an `Ltac` to include some test cases, as a means of light testing and never-outdated documentation.
Currently, when doing `Qed` on a `Goal`, an `Unnamed_thm` is registered. It would be better if nothing was registered, so that the files don't become bigger than needed, and that the `Search` command doesn't list all these `Unnamed_thm`s, which requires users either to set up filters in their IDEs, or to do `Abort` on all `Goals` instead of `Qed`.
In short: Here's one user who would find this improvement useful!

**Comment by Alizter** [comment]
@samuelgruetter Thanks, my phone must have autocorrected my typing

**Comment by SkySkimmer** [comment]
IMO if you don't want to save the proof then you shouldn't save it. Use `Succeed Qed. Abort.` instead.

**Comment by samuelgruetter** [comment]
Aah I wasn't aware of the `Succeed` command, thanks @SkySkimmer for sharing this trick :+1:

Depending on how much it complicates the code base, one might still consider making Qed-after-Goal an alias for `Succeed Qed. Abort.` though?

**Comment by SkySkimmer** [comment]
Having Qed do different things for Goal vs everything else sounds confusing
